==100093== Memcheck, a memory error detector
==100093== Copyright (C) 2002-2022, and GNU GPL'd, by Julian Seward et al.
==100093== Using Valgrind-3.22.0 and LibVEX; rerun with -h for copyright info
==100093== Command: /home/marija/Desktop/2024_Analysis_12-Uno/valgrind/memcheck/../../12-UNO/UNO/build/run_tests
==100093== Parent PID: 99624
==100093== 
==100093== Conditional jump or move depends on uninitialised value(s)
==100093==    at 0x13D826: Catch::RunContext::handleExpr(Catch::AssertionInfo const&, Catch::ITransientExpression const&, Catch::AssertionReaction&) (catch.hpp:13049)
==100093==    by 0x12C627: Catch::AssertionHandler::handleExpr(Catch::ITransientExpression const&) (catch.hpp:8230)
==100093==    by 0x202B03: C_A_T_C_H_T_E_S_T_0() (test_pile.cpp:76)
==100093==    by 0x14415D: Catch::TestInvokerAsFunction::invoke() const (catch.hpp:14330)
==100093==    by 0x143394: Catch::TestCase::invoke() const (catch.hpp:14169)
==100093==    by 0x13D69F: Catch::RunContext::invokeActiveTestCase() (catch.hpp:13025)
==100093==    by 0x13D3C4: Catch::RunContext::runCurrentTest(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >&) (catch.hpp:12998)
==100093==    by 0x13BE50: Catch::RunContext::runTest(Catch::TestCase const&) (catch.hpp:12759)
==100093==    by 0x13F045: Catch::(anonymous namespace)::TestGroup::execute() (catch.hpp:13352)
==100093==    by 0x1404AC: Catch::Session::runInternal() (catch.hpp:13562)
==100093==    by 0x140186: Catch::Session::run() (catch.hpp:13518)
==100093==    by 0x1812EE: int Catch::Session::run<char>(int, char const* const*) (catch.hpp:13236)
==100093==  Uninitialised value was created by a heap allocation
==100093==    at 0x4846FA3: operator new(unsigned long) (in /usr/libexec/valgrind/vgpreload_memcheck-amd64-linux.so)
==100093==    by 0x2029F5: C_A_T_C_H_T_E_S_T_0() (test_pile.cpp:73)
==100093==    by 0x14415D: Catch::TestInvokerAsFunction::invoke() const (catch.hpp:14330)
==100093==    by 0x143394: Catch::TestCase::invoke() const (catch.hpp:14169)
==100093==    by 0x13D69F: Catch::RunContext::invokeActiveTestCase() (catch.hpp:13025)
==100093==    by 0x13D3C4: Catch::RunContext::runCurrentTest(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >&) (catch.hpp:12998)
==100093==    by 0x13BE50: Catch::RunContext::runTest(Catch::TestCase const&) (catch.hpp:12759)
==100093==    by 0x13F045: Catch::(anonymous namespace)::TestGroup::execute() (catch.hpp:13352)
==100093==    by 0x1404AC: Catch::Session::runInternal() (catch.hpp:13562)
==100093==    by 0x140186: Catch::Session::run() (catch.hpp:13518)
==100093==    by 0x1812EE: int Catch::Session::run<char>(int, char const* const*) (catch.hpp:13236)
==100093==    by 0x15735F: main (main.cpp:8)
==100093== 
==100093== Conditional jump or move depends on uninitialised value(s)
==100093==    at 0x13D826: Catch::RunContext::handleExpr(Catch::AssertionInfo const&, Catch::ITransientExpression const&, Catch::AssertionReaction&) (catch.hpp:13049)
==100093==    by 0x12C627: Catch::AssertionHandler::handleExpr(Catch::ITransientExpression const&) (catch.hpp:8230)
==100093==    by 0x2033C0: C_A_T_C_H_T_E_S_T_0() (test_pile.cpp:132)
==100093==    by 0x14415D: Catch::TestInvokerAsFunction::invoke() const (catch.hpp:14330)
==100093==    by 0x143394: Catch::TestCase::invoke() const (catch.hpp:14169)
==100093==    by 0x13D69F: Catch::RunContext::invokeActiveTestCase() (catch.hpp:13025)
==100093==    by 0x13D3C4: Catch::RunContext::runCurrentTest(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >&) (catch.hpp:12998)
==100093==    by 0x13BE50: Catch::RunContext::runTest(Catch::TestCase const&) (catch.hpp:12759)
==100093==    by 0x13F045: Catch::(anonymous namespace)::TestGroup::execute() (catch.hpp:13352)
==100093==    by 0x1404AC: Catch::Session::runInternal() (catch.hpp:13562)
==100093==    by 0x140186: Catch::Session::run() (catch.hpp:13518)
==100093==    by 0x1812EE: int Catch::Session::run<char>(int, char const* const*) (catch.hpp:13236)
==100093==  Uninitialised value was created by a heap allocation
==100093==    at 0x4846FA3: operator new(unsigned long) (in /usr/libexec/valgrind/vgpreload_memcheck-amd64-linux.so)
==100093==    by 0x235EF2: Pile::lastCard() const (pile.cpp:31)
==100093==    by 0x203193: C_A_T_C_H_T_E_S_T_0() (test_pile.cpp:127)
==100093==    by 0x14415D: Catch::TestInvokerAsFunction::invoke() const (catch.hpp:14330)
==100093==    by 0x143394: Catch::TestCase::invoke() const (catch.hpp:14169)
==100093==    by 0x13D69F: Catch::RunContext::invokeActiveTestCase() (catch.hpp:13025)
==100093==    by 0x13D3C4: Catch::RunContext::runCurrentTest(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >&) (catch.hpp:12998)
==100093==    by 0x13BE50: Catch::RunContext::runTest(Catch::TestCase const&) (catch.hpp:12759)
==100093==    by 0x13F045: Catch::(anonymous namespace)::TestGroup::execute() (catch.hpp:13352)
==100093==    by 0x1404AC: Catch::Session::runInternal() (catch.hpp:13562)
==100093==    by 0x140186: Catch::Session::run() (catch.hpp:13518)
==100093==    by 0x1812EE: int Catch::Session::run<char>(int, char const* const*) (catch.hpp:13236)
==100093== 
==100093== 
==100093== HEAP SUMMARY:
==100093==     in use at exit: 2,452,496 bytes in 24,368 blocks
==100093==   total heap usage: 326,150 allocs, 301,782 frees, 33,966,526 bytes allocated
==100093== 
==100093== 16 bytes in 1 blocks are still reachable in loss record 490 of 7,502
==100093==    at 0x4846FA3: operator new(unsigned long) (in /usr/libexec/valgrind/vgpreload_memcheck-amd64-linux.so)
==100093==    by 0x5D23C4F: ??? (in /home/marija/Qt/6.5.3/gcc_64/lib/libQt6Core.so.6.5.3)
==100093==    by 0x5CA44ED: QThread::currentThread() (in /home/marija/Qt/6.5.3/gcc_64/lib/libQt6Core.so.6.5.3)
==100093==    by 0x5B9019E: QCoreApplicationPrivate::QCoreApplicationPrivate(int&, char**) (in /home/marija/Qt/6.5.3/gcc_64/lib/libQt6Core.so.6.5.3)
==100093==    by 0x522B858: QGuiApplicationPrivate::QGuiApplicationPrivate(int&, char**) (in /home/marija/Qt/6.5.3/gcc_64/lib/libQt6Gui.so.6.5.3)
==100093==    by 0x5237C2D: QGuiApplication::QGuiApplication(int&, char**, int) (in /home/marija/Qt/6.5.3/gcc_64/lib/libQt6Gui.so.6.5.3)
==100093==    by 0x157332: main (main.cpp:7)
==100093== 
==100093== 20 bytes in 1 blocks are indirectly lost in loss record 1,619 of 7,502
==100093==    at 0x4846828: malloc (in /usr/libexec/valgrind/vgpreload_memcheck-amd64-linux.so)
==100093==    by 0x5CBB860: QArrayData::allocate(QArrayData**, long long, long long, long long, QArrayData::AllocationOption) (in /home/marija/Qt/6.5.3/gcc_64/lib/libQt6Core.so.6.5.3)
==100093==    by 0x5C89D2E: QString::QString(long long, Qt::Initialization) (in /home/marija/Qt/6.5.3/gcc_64/lib/libQt6Core.so.6.5.3)
==100093==    by 0x5C9E3A9: ??? (in /home/marija/Qt/6.5.3/gcc_64/lib/libQt6Core.so.6.5.3)
==100093==    by 0x5C84613: QString::fromUtf8(QByteArrayView) (in /home/marija/Qt/6.5.3/gcc_64/lib/libQt6Core.so.6.5.3)
==100093==    by 0x1FDC5D: QString::operator=(char const*) (qstring.h:1035)
==100093==    by 0x1FC08C: C_A_T_C_H_T_E_S_T_0() (test_deck.cpp:159)
==100093==    by 0x14415D: Catch::TestInvokerAsFunction::invoke() const (catch.hpp:14330)
==100093==    by 0x143394: Catch::TestCase::invoke() const (catch.hpp:14169)
==100093==    by 0x13D69F: Catch::RunContext::invokeActiveTestCase() (catch.hpp:13025)
==100093==    by 0x13D3C4: Catch::RunContext::runCurrentTest(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >&) (catch.hpp:12998)
==100093==    by 0x13BE50: Catch::RunContext::runTest(Catch::TestCase const&) (catch.hpp:12759)
==100093== 
==100093== 24 bytes in 1 blocks are indirectly lost in loss record 2,206 of 7,502
==100093==    at 0x4846FA3: operator new(unsigned long) (in /usr/libexec/valgrind/vgpreload_memcheck-amd64-linux.so)
==100093==    by 0x206A2E: C_A_T_C_H_T_E_S_T_0() (test_player.cpp:66)
==100093==    by 0x14415D: Catch::TestInvokerAsFunction::invoke() const (catch.hpp:14330)
==100093==    by 0x143394: Catch::TestCase::invoke() const (catch.hpp:14169)
==100093==    by 0x13D69F: Catch::RunContext::invokeActiveTestCase() (catch.hpp:13025)
==100093==    by 0x13D3C4: Catch::RunContext::runCurrentTest(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >&) (catch.hpp:12998)
==100093==    by 0x13BE50: Catch::RunContext::runTest(Catch::TestCase const&) (catch.hpp:12759)
==100093==    by 0x13F045: Catch::(anonymous namespace)::TestGroup::execute() (catch.hpp:13352)
==100093==    by 0x1404AC: Catch::Session::runInternal() (catch.hpp:13562)
==100093==    by 0x140186: Catch::Session::run() (catch.hpp:13518)
==100093==    by 0x1812EE: int Catch::Session::run<char>(int, char const* const*) (catch.hpp:13236)
==100093==    by 0x15735F: main (main.cpp:8)
==100093== 
==100093== 24 bytes in 1 blocks are indirectly lost in loss record 2,207 of 7,502
==100093==    at 0x4846FA3: operator new(unsigned long) (in /usr/libexec/valgrind/vgpreload_memcheck-amd64-linux.so)
==100093==    by 0x206A54: C_A_T_C_H_T_E_S_T_0() (test_player.cpp:67)
==100093==    by 0x14415D: Catch::TestInvokerAsFunction::invoke() const (catch.hpp:14330)
==100093==    by 0x143394: Catch::TestCase::invoke() const (catch.hpp:14169)
==100093==    by 0x13D69F: Catch::RunContext::invokeActiveTestCase() (catch.hpp:13025)
==100093==    by 0x13D3C4: Catch::RunContext::runCurrentTest(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >&) (catch.hpp:12998)
==100093==    by 0x13BE50: Catch::RunContext::runTest(Catch::TestCase const&) (catch.hpp:12759)
==100093==    by 0x13F045: Catch::(anonymous namespace)::TestGroup::execute() (catch.hpp:13352)
==100093==    by 0x1404AC: Catch::Session::runInternal() (catch.hpp:13562)
==100093==    by 0x140186: Catch::Session::run() (catch.hpp:13518)
==100093==    by 0x1812EE: int Catch::Session::run<char>(int, char const* const*) (catch.hpp:13236)
==100093==    by 0x15735F: main (main.cpp:8)
==100093== 
==100093== 24 bytes in 1 blocks are indirectly lost in loss record 2,208 of 7,502
==100093==    at 0x4846FA3: operator new(unsigned long) (in /usr/libexec/valgrind/vgpreload_memcheck-amd64-linux.so)
==100093==    by 0x206A75: C_A_T_C_H_T_E_S_T_0() (test_player.cpp:68)
==100093==    by 0x14415D: Catch::TestInvokerAsFunction::invoke() const (catch.hpp:14330)
==100093==    by 0x143394: Catch::TestCase::invoke() const (catch.hpp:14169)
==100093==    by 0x13D69F: Catch::RunContext::invokeActiveTestCase() (catch.hpp:13025)
==100093==    by 0x13D3C4: Catch::RunContext::runCurrentTest(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >&) (catch.hpp:12998)
==100093==    by 0x13BE50: Catch::RunContext::runTest(Catch::TestCase const&) (catch.hpp:12759)
==100093==    by 0x13F045: Catch::(anonymous namespace)::TestGroup::execute() (catch.hpp:13352)
==100093==    by 0x1404AC: Catch::Session::runInternal() (catch.hpp:13562)
==100093==    by 0x140186: Catch::Session::run() (catch.hpp:13518)
==100093==    by 0x1812EE: int Catch::Session::run<char>(int, char const* const*) (catch.hpp:13236)
==100093==    by 0x15735F: main (main.cpp:8)
==100093== 
==100093== 24 bytes in 1 blocks are indirectly lost in loss record 2,209 of 7,502
==100093==    at 0x4846FA3: operator new(unsigned long) (in /usr/libexec/valgrind/vgpreload_memcheck-amd64-linux.so)
==100093==    by 0x208A4D: C_A_T_C_H_T_E_S_T_0() (test_player.cpp:323)
==100093==    by 0x14415D: Catch::TestInvokerAsFunction::invoke() const (catch.hpp:14330)
==100093==    by 0x143394: Catch::TestCase::invoke() const (catch.hpp:14169)
==100093==    by 0x13D69F: Catch::RunContext::invokeActiveTestCase() (catch.hpp:13025)
==100093==    by 0x13D3C4: Catch::RunContext::runCurrentTest(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >&) (catch.hpp:12998)
==100093==    by 0x13BE50: Catch::RunContext::runTest(Catch::TestCase const&) (catch.hpp:12759)
==100093==    by 0x13F045: Catch::(anonymous namespace)::TestGroup::execute() (catch.hpp:13352)
==100093==    by 0x1404AC: Catch::Session::runInternal() (catch.hpp:13562)
==100093==    by 0x140186: Catch::Session::run() (catch.hpp:13518)
==100093==    by 0x1812EE: int Catch::Session::run<char>(int, char const* const*) (catch.hpp:13236)
==100093==    by 0x15735F: main (main.cpp:8)
==100093== 
==100093== 24 bytes in 1 blocks are indirectly lost in loss record 2,210 of 7,502
==100093==    at 0x4846FA3: operator new(unsigned long) (in /usr/libexec/valgrind/vgpreload_memcheck-amd64-linux.so)
==100093==    by 0x208A73: C_A_T_C_H_T_E_S_T_0() (test_player.cpp:324)
==100093==    by 0x14415D: Catch::TestInvokerAsFunction::invoke() const (catch.hpp:14330)
==100093==    by 0x143394: Catch::TestCase::invoke() const (catch.hpp:14169)
==100093==    by 0x13D69F: Catch::RunContext::invokeActiveTestCase() (catch.hpp:13025)
==100093==    by 0x13D3C4: Catch::RunContext::runCurrentTest(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >&) (catch.hpp:12998)
==100093==    by 0x13BE50: Catch::RunContext::runTest(Catch::TestCase const&) (catch.hpp:12759)
==100093==    by 0x13F045: Catch::(anonymous namespace)::TestGroup::execute() (catch.hpp:13352)
==100093==    by 0x1404AC: Catch::Session::runInternal() (catch.hpp:13562)
==100093==    by 0x140186: Catch::Session::run() (catch.hpp:13518)
==100093==    by 0x1812EE: int Catch::Session::run<char>(int, char const* const*) (catch.hpp:13236)
==100093==    by 0x15735F: main (main.cpp:8)
==100093== 
==100093== 24 bytes in 1 blocks are indirectly lost in loss record 2,211 of 7,502
==100093==    at 0x4846FA3: operator new(unsigned long) (in /usr/libexec/valgrind/vgpreload_memcheck-amd64-linux.so)
==100093==    by 0x208A99: C_A_T_C_H_T_E_S_T_0() (test_player.cpp:325)
==100093==    by 0x14415D: Catch::TestInvokerAsFunction::invoke() const (catch.hpp:14330)
==100093==    by 0x143394: Catch::TestCase::invoke() const (catch.hpp:14169)
==100093==    by 0x13D69F: Catch::RunContext::invokeActiveTestCase() (catch.hpp:13025)
==100093==    by 0x13D3C4: Catch::RunContext::runCurrentTest(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >&) (catch.hpp:12998)
==100093==    by 0x13BE50: Catch::RunContext::runTest(Catch::TestCase const&) (catch.hpp:12759)
==100093==    by 0x13F045: Catch::(anonymous namespace)::TestGroup::execute() (catch.hpp:13352)
==100093==    by 0x1404AC: Catch::Session::runInternal() (catch.hpp:13562)
==100093==    by 0x140186: Catch::Session::run() (catch.hpp:13518)
==100093==    by 0x1812EE: int Catch::Session::run<char>(int, char const* const*) (catch.hpp:13236)
==100093==    by 0x15735F: main (main.cpp:8)
==100093== 
==100093== 24 bytes in 1 blocks are indirectly lost in loss record 2,212 of 7,502
==100093==    at 0x4846FA3: operator new(unsigned long) (in /usr/libexec/valgrind/vgpreload_memcheck-amd64-linux.so)
==100093==    by 0x20CCEF: C_A_T_C_H_T_E_S_T_0() (test_game.cpp:19)
==100093==    by 0x14415D: Catch::TestInvokerAsFunction::invoke() const (catch.hpp:14330)
==100093==    by 0x143394: Catch::TestCase::invoke() const (catch.hpp:14169)
==100093==    by 0x13D69F: Catch::RunContext::invokeActiveTestCase() (catch.hpp:13025)
==100093==    by 0x13D3C4: Catch::RunContext::runCurrentTest(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >&) (catch.hpp:12998)
==100093==    by 0x13BE50: Catch::RunContext::runTest(Catch::TestCase const&) (catch.hpp:12759)
==100093==    by 0x13F045: Catch::(anonymous namespace)::TestGroup::execute() (catch.hpp:13352)
==100093==    by 0x1404AC: Catch::Session::runInternal() (catch.hpp:13562)
==100093==    by 0x140186: Catch::Session::run() (catch.hpp:13518)
==100093==    by 0x1812EE: int Catch::Session::run<char>(int, char const* const*) (catch.hpp:13236)
==100093==    by 0x15735F: main (main.cpp:8)
==100093== 
==100093== 24 bytes in 1 blocks are definitely lost in loss record 2,213 of 7,502
==100093==    at 0x4846FA3: operator new(unsigned long) (in /usr/libexec/valgrind/vgpreload_memcheck-amd64-linux.so)
==100093==    by 0x235EF2: Pile::lastCard() const (pile.cpp:31)
==100093==    by 0x203193: C_A_T_C_H_T_E_S_T_0() (test_pile.cpp:127)
==100093==    by 0x14415D: Catch::TestInvokerAsFunction::invoke() const (catch.hpp:14330)
==100093==    by 0x143394: Catch::TestCase::invoke() const (catch.hpp:14169)
==100093==    by 0x13D69F: Catch::RunContext::invokeActiveTestCase() (catch.hpp:13025)
==100093==    by 0x13D3C4: Catch::RunContext::runCurrentTest(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >&) (catch.hpp:12998)
==100093==    by 0x13BE50: Catch::RunContext::runTest(Catch::TestCase const&) (catch.hpp:12759)
==100093==    by 0x13F045: Catch::(anonymous namespace)::TestGroup::execute() (catch.hpp:13352)
==100093==    by 0x1404AC: Catch::Session::runInternal() (catch.hpp:13562)
==100093==    by 0x140186: Catch::Session::run() (catch.hpp:13518)
==100093==    by 0x1812EE: int Catch::Session::run<char>(int, char const* const*) (catch.hpp:13236)
==100093== 
==100093== 24 bytes in 1 blocks are definitely lost in loss record 2,214 of 7,502
==100093==    at 0x4846FA3: operator new(unsigned long) (in /usr/libexec/valgrind/vgpreload_memcheck-amd64-linux.so)
==100093==    by 0x204134: C_A_T_C_H_T_E_S_T_0() (test_pile.cpp:228)
==100093==    by 0x14415D: Catch::TestInvokerAsFunction::invoke() const (catch.hpp:14330)
==100093==    by 0x143394: Catch::TestCase::invoke() const (catch.hpp:14169)
==100093==    by 0x13D69F: Catch::RunContext::invokeActiveTestCase() (catch.hpp:13025)
==100093==    by 0x13D3C4: Catch::RunContext::runCurrentTest(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >&) (catch.hpp:12998)
==100093==    by 0x13BE50: Catch::RunContext::runTest(Catch::TestCase const&) (catch.hpp:12759)
==100093==    by 0x13F045: Catch::(anonymous namespace)::TestGroup::execute() (catch.hpp:13352)
==100093==    by 0x1404AC: Catch::Session::runInternal() (catch.hpp:13562)
==100093==    by 0x140186: Catch::Session::run() (catch.hpp:13518)
==100093==    by 0x1812EE: int Catch::Session::run<char>(int, char const* const*) (catch.hpp:13236)
==100093==    by 0x15735F: main (main.cpp:8)
==100093== 
==100093== 24 bytes in 1 blocks are definitely lost in loss record 2,215 of 7,502
==100093==    at 0x4846FA3: operator new(unsigned long) (in /usr/libexec/valgrind/vgpreload_memcheck-amd64-linux.so)
==100093==    by 0x206310: C_A_T_C_H_T_E_S_T_0() (test_player.cpp:14)
==100093==    by 0x14415D: Catch::TestInvokerAsFunction::invoke() const (catch.hpp:14330)
==100093==    by 0x143394: Catch::TestCase::invoke() const (catch.hpp:14169)
==100093==    by 0x13D69F: Catch::RunContext::invokeActiveTestCase() (catch.hpp:13025)
==100093==    by 0x13D3C4: Catch::RunContext::runCurrentTest(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >&) (catch.hpp:12998)
==100093==    by 0x13BE50: Catch::RunContext::runTest(Catch::TestCase const&) (catch.hpp:12759)
==100093==    by 0x13F045: Catch::(anonymous namespace)::TestGroup::execute() (catch.hpp:13352)
==100093==    by 0x1404AC: Catch::Session::runInternal() (catch.hpp:13562)
==100093==    by 0x140186: Catch::Session::run() (catch.hpp:13518)
==100093==    by 0x1812EE: int Catch::Session::run<char>(int, char const* const*) (catch.hpp:13236)
==100093==    by 0x15735F: main (main.cpp:8)
==100093== 
==100093== 24 bytes in 1 blocks are definitely lost in loss record 2,216 of 7,502
==100093==    at 0x4846FA3: operator new(unsigned long) (in /usr/libexec/valgrind/vgpreload_memcheck-amd64-linux.so)
==100093==    by 0x206A08: C_A_T_C_H_T_E_S_T_0() (test_player.cpp:65)
==100093==    by 0x14415D: Catch::TestInvokerAsFunction::invoke() const (catch.hpp:14330)
==100093==    by 0x143394: Catch::TestCase::invoke() const (catch.hpp:14169)
==100093==    by 0x13D69F: Catch::RunContext::invokeActiveTestCase() (catch.hpp:13025)
==100093==    by 0x13D3C4: Catch::RunContext::runCurrentTest(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >&) (catch.hpp:12998)
==100093==    by 0x13BE50: Catch::RunContext::runTest(Catch::TestCase const&) (catch.hpp:12759)
==100093==    by 0x13F045: Catch::(anonymous namespace)::TestGroup::execute() (catch.hpp:13352)
==100093==    by 0x1404AC: Catch::Session::runInternal() (catch.hpp:13562)
==100093==    by 0x140186: Catch::Session::run() (catch.hpp:13518)
==100093==    by 0x1812EE: int Catch::Session::run<char>(int, char const* const*) (catch.hpp:13236)
==100093==    by 0x15735F: main (main.cpp:8)
==100093== 
==100093== 24 bytes in 1 blocks are definitely lost in loss record 2,217 of 7,502
==100093==    at 0x4846FA3: operator new(unsigned long) (in /usr/libexec/valgrind/vgpreload_memcheck-amd64-linux.so)
==100093==    by 0x2096FC: C_A_T_C_H_T_E_S_T_0() (test_player.cpp:422)
==100093==    by 0x14415D: Catch::TestInvokerAsFunction::invoke() const (catch.hpp:14330)
==100093==    by 0x143394: Catch::TestCase::invoke() const (catch.hpp:14169)
==100093==    by 0x13D69F: Catch::RunContext::invokeActiveTestCase() (catch.hpp:13025)
==100093==    by 0x13D3C4: Catch::RunContext::runCurrentTest(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >&) (catch.hpp:12998)
==100093==    by 0x13BE50: Catch::RunContext::runTest(Catch::TestCase const&) (catch.hpp:12759)
==100093==    by 0x13F045: Catch::(anonymous namespace)::TestGroup::execute() (catch.hpp:13352)
==100093==    by 0x1404AC: Catch::Session::runInternal() (catch.hpp:13562)
==100093==    by 0x140186: Catch::Session::run() (catch.hpp:13518)
==100093==    by 0x1812EE: int Catch::Session::run<char>(int, char const* const*) (catch.hpp:13236)
==100093==    by 0x15735F: main (main.cpp:8)
==100093== 
==100093== 24 bytes in 1 blocks are definitely lost in loss record 2,218 of 7,502
==100093==    at 0x4846FA3: operator new(unsigned long) (in /usr/libexec/valgrind/vgpreload_memcheck-amd64-linux.so)
==100093==    by 0x20D4FB: C_A_T_C_H_T_E_S_T_0() (test_game.cpp:80)
==100093==    by 0x14415D: Catch::TestInvokerAsFunction::invoke() const (catch.hpp:14330)
==100093==    by 0x143394: Catch::TestCase::invoke() const (catch.hpp:14169)
==100093==    by 0x13D69F: Catch::RunContext::invokeActiveTestCase() (catch.hpp:13025)
==100093==    by 0x13D3C4: Catch::RunContext::runCurrentTest(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >&) (catch.hpp:12998)
==100093==    by 0x13BE50: Catch::RunContext::runTest(Catch::TestCase const&) (catch.hpp:12759)
==100093==    by 0x13F045: Catch::(anonymous namespace)::TestGroup::execute() (catch.hpp:13352)
==100093==    by 0x1404AC: Catch::Session::runInternal() (catch.hpp:13562)
==100093==    by 0x140186: Catch::Session::run() (catch.hpp:13518)
==100093==    by 0x1812EE: int Catch::Session::run<char>(int, char const* const*) (catch.hpp:13236)
==100093==    by 0x15735F: main (main.cpp:8)
==100093== 
==100093== 32 bytes in 1 blocks are indirectly lost in loss record 3,516 of 7,502
==100093==    at 0x4846828: malloc (in /usr/libexec/valgrind/vgpreload_memcheck-amd64-linux.so)
==100093==    by 0x5CBB860: QArrayData::allocate(QArrayData**, long long, long long, long long, QArrayData::AllocationOption) (in /home/marija/Qt/6.5.3/gcc_64/lib/libQt6Core.so.6.5.3)
==100093==    by 0x5C89D2E: QString::QString(long long, Qt::Initialization) (in /home/marija/Qt/6.5.3/gcc_64/lib/libQt6Core.so.6.5.3)
==100093==    by 0x5C9E3A9: ??? (in /home/marija/Qt/6.5.3/gcc_64/lib/libQt6Core.so.6.5.3)
==100093==    by 0x5C84613: QString::fromUtf8(QByteArrayView) (in /home/marija/Qt/6.5.3/gcc_64/lib/libQt6Core.so.6.5.3)
==100093==    by 0x1FDC5D: QString::operator=(char const*) (qstring.h:1035)
==100093==    by 0x1FC059: C_A_T_C_H_T_E_S_T_0() (test_deck.cpp:156)
==100093==    by 0x14415D: Catch::TestInvokerAsFunction::invoke() const (catch.hpp:14330)
==100093==    by 0x143394: Catch::TestCase::invoke() const (catch.hpp:14169)
==100093==    by 0x13D69F: Catch::RunContext::invokeActiveTestCase() (catch.hpp:13025)
==100093==    by 0x13D3C4: Catch::RunContext::runCurrentTest(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >&) (catch.hpp:12998)
==100093==    by 0x13BE50: Catch::RunContext::runTest(Catch::TestCase const&) (catch.hpp:12759)
==100093== 
==100093== 34 bytes in 1 blocks are indirectly lost in loss record 3,530 of 7,502
==100093==    at 0x4846828: malloc (in /usr/libexec/valgrind/vgpreload_memcheck-amd64-linux.so)
==100093==    by 0x5CBB860: QArrayData::allocate(QArrayData**, long long, long long, long long, QArrayData::AllocationOption) (in /home/marija/Qt/6.5.3/gcc_64/lib/libQt6Core.so.6.5.3)
==100093==    by 0x20193E: QTypedArrayData<Card*>::allocate(long long, QArrayData::AllocationOption) (qarraydata.h:105)
==100093==    by 0x2331C0: QArrayDataPointer<Card*>::allocateGrow(QArrayDataPointer<Card*> const&, long long, QArrayData::GrowthPosition) (qarraydatapointer.h:352)
==100093==    by 0x232882: QArrayDataPointer<Card*>::reallocateAndGrow(QArrayData::GrowthPosition, long long, QArrayDataPointer<Card*>*) (qarraydatapointer.h:213)
==100093==    by 0x232B8E: QArrayDataPointer<Card*>::detachAndGrow(QArrayData::GrowthPosition, long long, Card* const**, QArrayDataPointer<Card*>*) (qarraydatapointer.h:194)
==100093==    by 0x23227C: void QtPrivate::QPodArrayOps<Card*>::emplace<Card*&>(long long, Card*&) (qarraydataops.h:174)
==100093==    by 0x23177B: Card*& QList<Card*>::emplaceBack<Card*&>(Card*&) (qlist.h:856)
==100093==    by 0x230E86: QList<Card*>::append(Card*) (qlist.h:433)
==100093==    by 0x230A7C: QList<Card*>::push_back(Card*) (qlist.h:651)
==100093==    by 0x235E7D: Pile::pushCard(Card*) (pile.cpp:17)
==100093==    by 0x20CD29: C_A_T_C_H_T_E_S_T_0() (test_game.cpp:20)
==100093== 
==100093== 48 bytes in 1 blocks are still reachable in loss record 3,900 of 7,502
==100093==    at 0x4846828: malloc (in /usr/libexec/valgrind/vgpreload_memcheck-amd64-linux.so)
==100093==    by 0x6BD5B09: g_malloc (in /usr/lib/x86_64-linux-gnu/libglib-2.0.so.0.8000.0)
==100093==    by 0x6BCE3A9: ??? (in /usr/lib/x86_64-linux-gnu/libglib-2.0.so.0.8000.0)
==100093==    by 0x6BCFBC7: ??? (in /usr/lib/x86_64-linux-gnu/libglib-2.0.so.0.8000.0)
==100093==    by 0x6BCFCF6: g_source_attach (in /usr/lib/x86_64-linux-gnu/libglib-2.0.so.0.8000.0)
==100093==    by 0xBA7D69A: ??? (in /usr/lib/x86_64-linux-gnu/libgio-2.0.so.0.8000.0)
==100093==    by 0xBA7D847: ??? (in /usr/lib/x86_64-linux-gnu/libgio-2.0.so.0.8000.0)
==100093==    by 0x6C04531: ??? (in /usr/lib/x86_64-linux-gnu/libglib-2.0.so.0.8000.0)
==100093==    by 0x6BFED91: ??? (in /usr/lib/x86_64-linux-gnu/libglib-2.0.so.0.8000.0)
==100093==    by 0x6587AA3: start_thread (pthread_create.c:447)
==100093==    by 0x6614A63: clone (clone.S:100)
==100093== 
==100093== 56 bytes in 1 blocks are still reachable in loss record 4,467 of 7,502
==100093==    at 0x4846828: malloc (in /usr/libexec/valgrind/vgpreload_memcheck-amd64-linux.so)
==100093==    by 0x6BD5B09: g_malloc (in /usr/lib/x86_64-linux-gnu/libglib-2.0.so.0.8000.0)
==100093==    by 0xB046F96: ??? (in /usr/lib/x86_64-linux-gnu/libgtk-3.so.0.2409.32)
==100093==    by 0xB050B02: gtk_entry_set_icon_from_pixbuf (in /usr/lib/x86_64-linux-gnu/libgtk-3.so.0.2409.32)
==100093==    by 0xB04559A: ??? (in /usr/lib/x86_64-linux-gnu/libgtk-3.so.0.2409.32)
==100093==    by 0xBBBC6C3: g_object_run_dispose (in /usr/lib/x86_64-linux-gnu/libgobject-2.0.so.0.8000.0)
==100093==    by 0xAEBCE7C: ??? (in /home/marija/Qt/6.5.3/gcc_64/plugins/platformthemes/libqgtk3.so)
==100093==    by 0xAEB31C4: ??? (in /home/marija/Qt/6.5.3/gcc_64/plugins/platformthemes/libqgtk3.so)
==100093==    by 0xAEBBF24: ??? (in /home/marija/Qt/6.5.3/gcc_64/plugins/platformthemes/libqgtk3.so)
==100093==    by 0x522B0C6: QGuiApplicationPrivate::~QGuiApplicationPrivate() (in /home/marija/Qt/6.5.3/gcc_64/lib/libQt6Gui.so.6.5.3)
==100093==    by 0x522B518: QGuiApplicationPrivate::~QGuiApplicationPrivate() (in /home/marija/Qt/6.5.3/gcc_64/lib/libQt6Gui.so.6.5.3)
==100093==    by 0x157380: main (main.cpp:7)
==100093== 
==100093== 56 bytes in 1 blocks are still reachable in loss record 4,469 of 7,502
==100093==    at 0x4846828: malloc (in /usr/libexec/valgrind/vgpreload_memcheck-amd64-linux.so)
==100093==    by 0x6BD5B09: g_malloc (in /usr/lib/x86_64-linux-gnu/libglib-2.0.so.0.8000.0)
==100093==    by 0xB046F96: ??? (in /usr/lib/x86_64-linux-gnu/libgtk-3.so.0.2409.32)
==100093==    by 0xB050B02: gtk_entry_set_icon_from_pixbuf (in /usr/lib/x86_64-linux-gnu/libgtk-3.so.0.2409.32)
==100093==    by 0xB0455B5: ??? (in /usr/lib/x86_64-linux-gnu/libgtk-3.so.0.2409.32)
==100093==    by 0xBBBC6C3: g_object_run_dispose (in /usr/lib/x86_64-linux-gnu/libgobject-2.0.so.0.8000.0)
==100093==    by 0xAEBCE7C: ??? (in /home/marija/Qt/6.5.3/gcc_64/plugins/platformthemes/libqgtk3.so)
==100093==    by 0xAEB31C4: ??? (in /home/marija/Qt/6.5.3/gcc_64/plugins/platformthemes/libqgtk3.so)
==100093==    by 0xAEBBF24: ??? (in /home/marija/Qt/6.5.3/gcc_64/plugins/platformthemes/libqgtk3.so)
==100093==    by 0x522B0C6: QGuiApplicationPrivate::~QGuiApplicationPrivate() (in /home/marija/Qt/6.5.3/gcc_64/lib/libQt6Gui.so.6.5.3)
==100093==    by 0x522B518: QGuiApplicationPrivate::~QGuiApplicationPrivate() (in /home/marija/Qt/6.5.3/gcc_64/lib/libQt6Gui.so.6.5.3)
==100093==    by 0x157380: main (main.cpp:7)
==100093== 
==100093== 66 bytes in 1 blocks are still reachable in loss record 4,777 of 7,502
==100093==    at 0x4846828: malloc (in /usr/libexec/valgrind/vgpreload_memcheck-amd64-linux.so)
==100093==    by 0x5CBB860: QArrayData::allocate(QArrayData**, long long, long long, long long, QArrayData::AllocationOption) (in /home/marija/Qt/6.5.3/gcc_64/lib/libQt6Core.so.6.5.3)
==100093==    by 0x5BECDE1: ??? (in /home/marija/Qt/6.5.3/gcc_64/lib/libQt6Core.so.6.5.3)
==100093==    by 0x5D2BDD7: QThreadStorageData::get() const (in /home/marija/Qt/6.5.3/gcc_64/lib/libQt6Core.so.6.5.3)
==100093==    by 0x54857A9: QFontCache::cleanup() (in /home/marija/Qt/6.5.3/gcc_64/lib/libQt6Gui.so.6.5.3)
==100093==    by 0x522B03F: QGuiApplicationPrivate::~QGuiApplicationPrivate() (in /home/marija/Qt/6.5.3/gcc_64/lib/libQt6Gui.so.6.5.3)
==100093==    by 0x522B518: QGuiApplicationPrivate::~QGuiApplicationPrivate() (in /home/marija/Qt/6.5.3/gcc_64/lib/libQt6Gui.so.6.5.3)
==100093==    by 0x157380: main (main.cpp:7)
==100093== 
==100093== 66 bytes in 1 blocks are indirectly lost in loss record 4,779 of 7,502
==100093==    at 0x4846828: malloc (in /usr/libexec/valgrind/vgpreload_memcheck-amd64-linux.so)
==100093==    by 0x5CBB860: QArrayData::allocate(QArrayData**, long long, long long, long long, QArrayData::AllocationOption) (in /home/marija/Qt/6.5.3/gcc_64/lib/libQt6Core.so.6.5.3)
==100093==    by 0x20193E: QTypedArrayData<Card*>::allocate(long long, QArrayData::AllocationOption) (qarraydata.h:105)
==100093==    by 0x2331C0: QArrayDataPointer<Card*>::allocateGrow(QArrayDataPointer<Card*> const&, long long, QArrayData::GrowthPosition) (qarraydatapointer.h:352)
==100093==    by 0x232882: QArrayDataPointer<Card*>::reallocateAndGrow(QArrayData::GrowthPosition, long long, QArrayDataPointer<Card*>*) (qarraydatapointer.h:213)
==100093==    by 0x232B8E: QArrayDataPointer<Card*>::detachAndGrow(QArrayData::GrowthPosition, long long, Card* const**, QArrayDataPointer<Card*>*) (qarraydatapointer.h:194)
==100093==    by 0x237163: QtPrivate::QCommonArrayOps<Card*>::growAppend(Card* const*, Card* const*) (qarraydataops.h:922)
==100093==    by 0x236F7B: QList<Card*>::append(QList<Card*>::const_iterator, QList<Card*>::const_iterator) (qlist.h:804)
==100093==    by 0x236DDE: QList<Card*>::append(QList<Card*> const&) (qlist.h:445)
==100093==    by 0x236A62: Player::setHand(QList<Card*> const&) (player.cpp:110)
==100093==    by 0x206B1D: C_A_T_C_H_T_E_S_T_0() (test_player.cpp:72)
==100093==    by 0x14415D: Catch::TestInvokerAsFunction::invoke() const (catch.hpp:14330)
==100093== 
==100093== 66 bytes in 1 blocks are indirectly lost in loss record 4,780 of 7,502
==100093==    at 0x484DB80: realloc (in /usr/libexec/valgrind/vgpreload_memcheck-amd64-linux.so)
==100093==    by 0x5CBB920: QArrayData::reallocateUnaligned(QArrayData*, void*, long long, long long, QArrayData::AllocationOption) (in /home/marija/Qt/6.5.3/gcc_64/lib/libQt6Core.so.6.5.3)
==100093==    by 0x2335FE: QTypedArrayData<Card*>::reallocateUnaligned(QTypedArrayData<Card*>*, Card**, long long, QArrayData::AllocationOption) (qarraydata.h:117)
==100093==    by 0x233000: QtPrivate::QPodArrayOps<Card*>::reallocate(long long, QArrayData::AllocationOption) (qarraydataops.h:259)
==100093==    by 0x232866: QArrayDataPointer<Card*>::reallocateAndGrow(QArrayData::GrowthPosition, long long, QArrayDataPointer<Card*>*) (qarraydatapointer.h:208)
==100093==    by 0x232B8E: QArrayDataPointer<Card*>::detachAndGrow(QArrayData::GrowthPosition, long long, Card* const**, QArrayDataPointer<Card*>*) (qarraydatapointer.h:194)
==100093==    by 0x23227C: void QtPrivate::QPodArrayOps<Card*>::emplace<Card*&>(long long, Card*&) (qarraydataops.h:174)
==100093==    by 0x23177B: Card*& QList<Card*>::emplaceBack<Card*&>(Card*&) (qlist.h:856)
==100093==    by 0x230E86: QList<Card*>::append(Card*) (qlist.h:433)
==100093==    by 0x230A7C: QList<Card*>::push_back(Card*) (qlist.h:651)
==100093==    by 0x235E7D: Pile::pushCard(Card*) (pile.cpp:17)
==100093==    by 0x208B00: C_A_T_C_H_T_E_S_T_0() (test_player.cpp:328)
==100093== 
==100093== 90 (32 direct, 58 indirect) bytes in 1 blocks are definitely lost in loss record 5,556 of 7,502
==100093==    at 0x4846FA3: operator new(unsigned long) (in /usr/libexec/valgrind/vgpreload_memcheck-amd64-linux.so)
==100093==    by 0x20CCA8: C_A_T_C_H_T_E_S_T_0() (test_game.cpp:17)
==100093==    by 0x14415D: Catch::TestInvokerAsFunction::invoke() const (catch.hpp:14330)
==100093==    by 0x143394: Catch::TestCase::invoke() const (catch.hpp:14169)
==100093==    by 0x13D69F: Catch::RunContext::invokeActiveTestCase() (catch.hpp:13025)
==100093==    by 0x13D3C4: Catch::RunContext::runCurrentTest(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >&) (catch.hpp:12998)
==100093==    by 0x13BE50: Catch::RunContext::runTest(Catch::TestCase const&) (catch.hpp:12759)
==100093==    by 0x13F045: Catch::(anonymous namespace)::TestGroup::execute() (catch.hpp:13352)
==100093==    by 0x1404AC: Catch::Session::runInternal() (catch.hpp:13562)
==100093==    by 0x140186: Catch::Session::run() (catch.hpp:13518)
==100093==    by 0x1812EE: int Catch::Session::run<char>(int, char const* const*) (catch.hpp:13236)
==100093==    by 0x15735F: main (main.cpp:8)
==100093== 
==100093== 96 bytes in 1 blocks are still reachable in loss record 5,588 of 7,502
==100093==    at 0x4846FA3: operator new(unsigned long) (in /usr/libexec/valgrind/vgpreload_memcheck-amd64-linux.so)
==100093==    by 0x5D2C994: QWaitCondition::QWaitCondition() (in /home/marija/Qt/6.5.3/gcc_64/lib/libQt6Core.so.6.5.3)
==100093==    by 0x5CA50A3: ??? (in /home/marija/Qt/6.5.3/gcc_64/lib/libQt6Core.so.6.5.3)
==100093==    by 0x5D23C5D: ??? (in /home/marija/Qt/6.5.3/gcc_64/lib/libQt6Core.so.6.5.3)
==100093==    by 0x5CA44ED: QThread::currentThread() (in /home/marija/Qt/6.5.3/gcc_64/lib/libQt6Core.so.6.5.3)
==100093==    by 0x5B9019E: QCoreApplicationPrivate::QCoreApplicationPrivate(int&, char**) (in /home/marija/Qt/6.5.3/gcc_64/lib/libQt6Core.so.6.5.3)
==100093==    by 0x522B858: QGuiApplicationPrivate::QGuiApplicationPrivate(int&, char**) (in /home/marija/Qt/6.5.3/gcc_64/lib/libQt6Gui.so.6.5.3)
==100093==    by 0x5237C2D: QGuiApplication::QGuiApplication(int&, char**, int) (in /home/marija/Qt/6.5.3/gcc_64/lib/libQt6Gui.so.6.5.3)
==100093==    by 0x157332: main (main.cpp:7)
==100093== 
==100093== 96 bytes in 4 blocks are definitely lost in loss record 6,419 of 7,502
==100093==    at 0x4846FA3: operator new(unsigned long) (in /usr/libexec/valgrind/vgpreload_memcheck-amd64-linux.so)
==100093==    by 0x22FFBC: Deck::Deck() (deck.cpp:41)
==100093==    by 0x1FBD15: C_A_T_C_H_T_E_S_T_0() (test_deck.cpp:135)
==100093==    by 0x14415D: Catch::TestInvokerAsFunction::invoke() const (catch.hpp:14330)
==100093==    by 0x143394: Catch::TestCase::invoke() const (catch.hpp:14169)
==100093==    by 0x13D69F: Catch::RunContext::invokeActiveTestCase() (catch.hpp:13025)
==100093==    by 0x13D3C4: Catch::RunContext::runCurrentTest(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >&) (catch.hpp:12998)
==100093==    by 0x13BE50: Catch::RunContext::runTest(Catch::TestCase const&) (catch.hpp:12759)
==100093==    by 0x13F045: Catch::(anonymous namespace)::TestGroup::execute() (catch.hpp:13352)
==100093==    by 0x1404AC: Catch::Session::runInternal() (catch.hpp:13562)
==100093==    by 0x140186: Catch::Session::run() (catch.hpp:13518)
==100093==    by 0x1812EE: int Catch::Session::run<char>(int, char const* const*) (catch.hpp:13236)
==100093== 
==100093== 96 bytes in 4 blocks are definitely lost in loss record 6,420 of 7,502
==100093==    at 0x4846FA3: operator new(unsigned long) (in /usr/libexec/valgrind/vgpreload_memcheck-amd64-linux.so)
==100093==    by 0x22FFF0: Deck::Deck() (deck.cpp:42)
==100093==    by 0x1FBD15: C_A_T_C_H_T_E_S_T_0() (test_deck.cpp:135)
==100093==    by 0x14415D: Catch::TestInvokerAsFunction::invoke() const (catch.hpp:14330)
==100093==    by 0x143394: Catch::TestCase::invoke() const (catch.hpp:14169)
==100093==    by 0x13D69F: Catch::RunContext::invokeActiveTestCase() (catch.hpp:13025)
==100093==    by 0x13D3C4: Catch::RunContext::runCurrentTest(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >&) (catch.hpp:12998)
==100093==    by 0x13BE50: Catch::RunContext::runTest(Catch::TestCase const&) (catch.hpp:12759)
==100093==    by 0x13F045: Catch::(anonymous namespace)::TestGroup::execute() (catch.hpp:13352)
==100093==    by 0x1404AC: Catch::Session::runInternal() (catch.hpp:13562)
==100093==    by 0x140186: Catch::Session::run() (catch.hpp:13518)
==100093==    by 0x1812EE: int Catch::Session::run<char>(int, char const* const*) (catch.hpp:13236)
==100093== 
==100093== 96 bytes in 4 blocks are definitely lost in loss record 6,421 of 7,502
==100093==    at 0x4846FA3: operator new(unsigned long) (in /usr/libexec/valgrind/vgpreload_memcheck-amd64-linux.so)
==100093==    by 0x22FFBC: Deck::Deck() (deck.cpp:41)
==100093==    by 0x1FC76C: C_A_T_C_H_T_E_S_T_0() (test_deck.cpp:206)
==100093==    by 0x14415D: Catch::TestInvokerAsFunction::invoke() const (catch.hpp:14330)
==100093==    by 0x143394: Catch::TestCase::invoke() const (catch.hpp:14169)
==100093==    by 0x13D69F: Catch::RunContext::invokeActiveTestCase() (catch.hpp:13025)
==100093==    by 0x13D3C4: Catch::RunContext::runCurrentTest(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >&) (catch.hpp:12998)
==100093==    by 0x13BE50: Catch::RunContext::runTest(Catch::TestCase const&) (catch.hpp:12759)
==100093==    by 0x13F045: Catch::(anonymous namespace)::TestGroup::execute() (catch.hpp:13352)
==100093==    by 0x1404AC: Catch::Session::runInternal() (catch.hpp:13562)
==100093==    by 0x140186: Catch::Session::run() (catch.hpp:13518)
==100093==    by 0x1812EE: int Catch::Session::run<char>(int, char const* const*) (catch.hpp:13236)
==100093== 
==100093== 96 bytes in 4 blocks are definitely lost in loss record 6,422 of 7,502
==100093==    at 0x4846FA3: operator new(unsigned long) (in /usr/libexec/valgrind/vgpreload_memcheck-amd64-linux.so)
==100093==    by 0x22FFF0: Deck::Deck() (deck.cpp:42)
==100093==    by 0x1FC76C: C_A_T_C_H_T_E_S_T_0() (test_deck.cpp:206)
==100093==    by 0x14415D: Catch::TestInvokerAsFunction::invoke() const (catch.hpp:14330)
==100093==    by 0x143394: Catch::TestCase::invoke() const (catch.hpp:14169)
==100093==    by 0x13D69F: Catch::RunContext::invokeActiveTestCase() (catch.hpp:13025)
==100093==    by 0x13D3C4: Catch::RunContext::runCurrentTest(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >&) (catch.hpp:12998)
==100093==    by 0x13BE50: Catch::RunContext::runTest(Catch::TestCase const&) (catch.hpp:12759)
==100093==    by 0x13F045: Catch::(anonymous namespace)::TestGroup::execute() (catch.hpp:13352)
==100093==    by 0x1404AC: Catch::Session::runInternal() (catch.hpp:13562)
==100093==    by 0x140186: Catch::Session::run() (catch.hpp:13518)
==100093==    by 0x1812EE: int Catch::Session::run<char>(int, char const* const*) (catch.hpp:13236)
==100093== 
==100093== 96 bytes in 4 blocks are definitely lost in loss record 6,423 of 7,502
==100093==    at 0x4846FA3: operator new(unsigned long) (in /usr/libexec/valgrind/vgpreload_memcheck-amd64-linux.so)
==100093==    by 0x22FFBC: Deck::Deck() (deck.cpp:41)
==100093==    by 0x1FC9B7: C_A_T_C_H_T_E_S_T_0() (test_deck.cpp:222)
==100093==    by 0x14415D: Catch::TestInvokerAsFunction::invoke() const (catch.hpp:14330)
==100093==    by 0x143394: Catch::TestCase::invoke() const (catch.hpp:14169)
==100093==    by 0x13D69F: Catch::RunContext::invokeActiveTestCase() (catch.hpp:13025)
==100093==    by 0x13D3C4: Catch::RunContext::runCurrentTest(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >&) (catch.hpp:12998)
==100093==    by 0x13BE50: Catch::RunContext::runTest(Catch::TestCase const&) (catch.hpp:12759)
==100093==    by 0x13F045: Catch::(anonymous namespace)::TestGroup::execute() (catch.hpp:13352)
==100093==    by 0x1404AC: Catch::Session::runInternal() (catch.hpp:13562)
==100093==    by 0x140186: Catch::Session::run() (catch.hpp:13518)
==100093==    by 0x1812EE: int Catch::Session::run<char>(int, char const* const*) (catch.hpp:13236)
==100093== 
==100093== 96 bytes in 4 blocks are definitely lost in loss record 6,424 of 7,502
==100093==    at 0x4846FA3: operator new(unsigned long) (in /usr/libexec/valgrind/vgpreload_memcheck-amd64-linux.so)
==100093==    by 0x22FFF0: Deck::Deck() (deck.cpp:42)
==100093==    by 0x1FC9B7: C_A_T_C_H_T_E_S_T_0() (test_deck.cpp:222)
==100093==    by 0x14415D: Catch::TestInvokerAsFunction::invoke() const (catch.hpp:14330)
==100093==    by 0x143394: Catch::TestCase::invoke() const (catch.hpp:14169)
==100093==    by 0x13D69F: Catch::RunContext::invokeActiveTestCase() (catch.hpp:13025)
==100093==    by 0x13D3C4: Catch::RunContext::runCurrentTest(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >&) (catch.hpp:12998)
==100093==    by 0x13BE50: Catch::RunContext::runTest(Catch::TestCase const&) (catch.hpp:12759)
==100093==    by 0x13F045: Catch::(anonymous namespace)::TestGroup::execute() (catch.hpp:13352)
==100093==    by 0x1404AC: Catch::Session::runInternal() (catch.hpp:13562)
==100093==    by 0x140186: Catch::Session::run() (catch.hpp:13518)
==100093==    by 0x1812EE: int Catch::Session::run<char>(int, char const* const*) (catch.hpp:13236)
==100093== 
==100093== 96 bytes in 4 blocks are definitely lost in loss record 6,425 of 7,502
==100093==    at 0x4846FA3: operator new(unsigned long) (in /usr/libexec/valgrind/vgpreload_memcheck-amd64-linux.so)
==100093==    by 0x22FFBC: Deck::Deck() (deck.cpp:41)
==100093==    by 0x20CD6F: C_A_T_C_H_T_E_S_T_0() (test_game.cpp:24)
==100093==    by 0x14415D: Catch::TestInvokerAsFunction::invoke() const (catch.hpp:14330)
==100093==    by 0x143394: Catch::TestCase::invoke() const (catch.hpp:14169)
==100093==    by 0x13D69F: Catch::RunContext::invokeActiveTestCase() (catch.hpp:13025)
==100093==    by 0x13D3C4: Catch::RunContext::runCurrentTest(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >&) (catch.hpp:12998)
==100093==    by 0x13BE50: Catch::RunContext::runTest(Catch::TestCase const&) (catch.hpp:12759)
==100093==    by 0x13F045: Catch::(anonymous namespace)::TestGroup::execute() (catch.hpp:13352)
==100093==    by 0x1404AC: Catch::Session::runInternal() (catch.hpp:13562)
==100093==    by 0x140186: Catch::Session::run() (catch.hpp:13518)
==100093==    by 0x1812EE: int Catch::Session::run<char>(int, char const* const*) (catch.hpp:13236)
==100093== 
==100093== 96 bytes in 4 blocks are definitely lost in loss record 6,426 of 7,502
==100093==    at 0x4846FA3: operator new(unsigned long) (in /usr/libexec/valgrind/vgpreload_memcheck-amd64-linux.so)
==100093==    by 0x22FFF0: Deck::Deck() (deck.cpp:42)
==100093==    by 0x20CD6F: C_A_T_C_H_T_E_S_T_0() (test_game.cpp:24)
==100093==    by 0x14415D: Catch::TestInvokerAsFunction::invoke() const (catch.hpp:14330)
==100093==    by 0x143394: Catch::TestCase::invoke() const (catch.hpp:14169)
==100093==    by 0x13D69F: Catch::RunContext::invokeActiveTestCase() (catch.hpp:13025)
==100093==    by 0x13D3C4: Catch::RunContext::runCurrentTest(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >&) (catch.hpp:12998)
==100093==    by 0x13BE50: Catch::RunContext::runTest(Catch::TestCase const&) (catch.hpp:12759)
==100093==    by 0x13F045: Catch::(anonymous namespace)::TestGroup::execute() (catch.hpp:13352)
==100093==    by 0x1404AC: Catch::Session::runInternal() (catch.hpp:13562)
==100093==    by 0x140186: Catch::Session::run() (catch.hpp:13518)
==100093==    by 0x1812EE: int Catch::Session::run<char>(int, char const* const*) (catch.hpp:13236)
==100093== 
==100093== 96 bytes in 4 blocks are definitely lost in loss record 6,427 of 7,502
==100093==    at 0x4846FA3: operator new(unsigned long) (in /usr/libexec/valgrind/vgpreload_memcheck-amd64-linux.so)
==100093==    by 0x22FFBC: Deck::Deck() (deck.cpp:41)
==100093==    by 0x20D4E3: C_A_T_C_H_T_E_S_T_0() (test_game.cpp:78)
==100093==    by 0x14415D: Catch::TestInvokerAsFunction::invoke() const (catch.hpp:14330)
==100093==    by 0x143394: Catch::TestCase::invoke() const (catch.hpp:14169)
==100093==    by 0x13D69F: Catch::RunContext::invokeActiveTestCase() (catch.hpp:13025)
==100093==    by 0x13D3C4: Catch::RunContext::runCurrentTest(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >&) (catch.hpp:12998)
==100093==    by 0x13BE50: Catch::RunContext::runTest(Catch::TestCase const&) (catch.hpp:12759)
==100093==    by 0x13F045: Catch::(anonymous namespace)::TestGroup::execute() (catch.hpp:13352)
==100093==    by 0x1404AC: Catch::Session::runInternal() (catch.hpp:13562)
==100093==    by 0x140186: Catch::Session::run() (catch.hpp:13518)
==100093==    by 0x1812EE: int Catch::Session::run<char>(int, char const* const*) (catch.hpp:13236)
==100093== 
==100093== 96 bytes in 4 blocks are definitely lost in loss record 6,428 of 7,502
==100093==    at 0x4846FA3: operator new(unsigned long) (in /usr/libexec/valgrind/vgpreload_memcheck-amd64-linux.so)
==100093==    by 0x22FFF0: Deck::Deck() (deck.cpp:42)
==100093==    by 0x20D4E3: C_A_T_C_H_T_E_S_T_0() (test_game.cpp:78)
==100093==    by 0x14415D: Catch::TestInvokerAsFunction::invoke() const (catch.hpp:14330)
==100093==    by 0x143394: Catch::TestCase::invoke() const (catch.hpp:14169)
==100093==    by 0x13D69F: Catch::RunContext::invokeActiveTestCase() (catch.hpp:13025)
==100093==    by 0x13D3C4: Catch::RunContext::runCurrentTest(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >&) (catch.hpp:12998)
==100093==    by 0x13BE50: Catch::RunContext::runTest(Catch::TestCase const&) (catch.hpp:12759)
==100093==    by 0x13F045: Catch::(anonymous namespace)::TestGroup::execute() (catch.hpp:13352)
==100093==    by 0x1404AC: Catch::Session::runInternal() (catch.hpp:13562)
==100093==    by 0x140186: Catch::Session::run() (catch.hpp:13518)
==100093==    by 0x1812EE: int Catch::Session::run<char>(int, char const* const*) (catch.hpp:13236)
==100093== 
==100093== 108 (56 direct, 52 indirect) bytes in 1 blocks are definitely lost in loss record 6,532 of 7,502
==100093==    at 0x4846FA3: operator new(unsigned long) (in /usr/libexec/valgrind/vgpreload_memcheck-amd64-linux.so)
==100093==    by 0x1FBFF6: C_A_T_C_H_T_E_S_T_0() (test_deck.cpp:155)
==100093==    by 0x14415D: Catch::TestInvokerAsFunction::invoke() const (catch.hpp:14330)
==100093==    by 0x143394: Catch::TestCase::invoke() const (catch.hpp:14169)
==100093==    by 0x13D69F: Catch::RunContext::invokeActiveTestCase() (catch.hpp:13025)
==100093==    by 0x13D3C4: Catch::RunContext::runCurrentTest(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >&) (catch.hpp:12998)
==100093==    by 0x13BE50: Catch::RunContext::runTest(Catch::TestCase const&) (catch.hpp:12759)
==100093==    by 0x13F045: Catch::(anonymous namespace)::TestGroup::execute() (catch.hpp:13352)
==100093==    by 0x1404AC: Catch::Session::runInternal() (catch.hpp:13562)
==100093==    by 0x140186: Catch::Session::run() (catch.hpp:13518)
==100093==    by 0x1812EE: int Catch::Session::run<char>(int, char const* const*) (catch.hpp:13236)
==100093==    by 0x15735F: main (main.cpp:8)
==100093== 
==100093== 152 bytes in 1 blocks are still reachable in loss record 6,789 of 7,502
==100093==    at 0x4846FA3: operator new(unsigned long) (in /usr/libexec/valgrind/vgpreload_memcheck-amd64-linux.so)
==100093==    by 0x5D23C01: ??? (in /home/marija/Qt/6.5.3/gcc_64/lib/libQt6Core.so.6.5.3)
==100093==    by 0x5CA44ED: QThread::currentThread() (in /home/marija/Qt/6.5.3/gcc_64/lib/libQt6Core.so.6.5.3)
==100093==    by 0x5B9019E: QCoreApplicationPrivate::QCoreApplicationPrivate(int&, char**) (in /home/marija/Qt/6.5.3/gcc_64/lib/libQt6Core.so.6.5.3)
==100093==    by 0x522B858: QGuiApplicationPrivate::QGuiApplicationPrivate(int&, char**) (in /home/marija/Qt/6.5.3/gcc_64/lib/libQt6Gui.so.6.5.3)
==100093==    by 0x5237C2D: QGuiApplication::QGuiApplication(int&, char**, int) (in /home/marija/Qt/6.5.3/gcc_64/lib/libQt6Gui.so.6.5.3)
==100093==    by 0x157332: main (main.cpp:7)
==100093== 
==100093== 170 (32 direct, 138 indirect) bytes in 1 blocks are definitely lost in loss record 6,848 of 7,502
==100093==    at 0x4846FA3: operator new(unsigned long) (in /usr/libexec/valgrind/vgpreload_memcheck-amd64-linux.so)
==100093==    by 0x208A31: C_A_T_C_H_T_E_S_T_0() (test_player.cpp:321)
==100093==    by 0x14415D: Catch::TestInvokerAsFunction::invoke() const (catch.hpp:14330)
==100093==    by 0x143394: Catch::TestCase::invoke() const (catch.hpp:14169)
==100093==    by 0x13D69F: Catch::RunContext::invokeActiveTestCase() (catch.hpp:13025)
==100093==    by 0x13D3C4: Catch::RunContext::runCurrentTest(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >&) (catch.hpp:12998)
==100093==    by 0x13BE50: Catch::RunContext::runTest(Catch::TestCase const&) (catch.hpp:12759)
==100093==    by 0x13F045: Catch::(anonymous namespace)::TestGroup::execute() (catch.hpp:13352)
==100093==    by 0x1404AC: Catch::Session::runInternal() (catch.hpp:13562)
==100093==    by 0x140186: Catch::Session::run() (catch.hpp:13518)
==100093==    by 0x1812EE: int Catch::Session::run<char>(int, char const* const*) (catch.hpp:13236)
==100093==    by 0x15735F: main (main.cpp:8)
==100093== 
==100093== 200 bytes in 1 blocks are still reachable in loss record 6,917 of 7,502
==100093==    at 0x4846FA3: operator new(unsigned long) (in /usr/libexec/valgrind/vgpreload_memcheck-amd64-linux.so)
==100093==    by 0x5CA5047: ??? (in /home/marija/Qt/6.5.3/gcc_64/lib/libQt6Core.so.6.5.3)
==100093==    by 0x5D23C5D: ??? (in /home/marija/Qt/6.5.3/gcc_64/lib/libQt6Core.so.6.5.3)
==100093==    by 0x5CA44ED: QThread::currentThread() (in /home/marija/Qt/6.5.3/gcc_64/lib/libQt6Core.so.6.5.3)
==100093==    by 0x5B9019E: QCoreApplicationPrivate::QCoreApplicationPrivate(int&, char**) (in /home/marija/Qt/6.5.3/gcc_64/lib/libQt6Core.so.6.5.3)
==100093==    by 0x522B858: QGuiApplicationPrivate::QGuiApplicationPrivate(int&, char**) (in /home/marija/Qt/6.5.3/gcc_64/lib/libQt6Gui.so.6.5.3)
==100093==    by 0x5237C2D: QGuiApplication::QGuiApplication(int&, char**, int) (in /home/marija/Qt/6.5.3/gcc_64/lib/libQt6Gui.so.6.5.3)
==100093==    by 0x157332: main (main.cpp:7)
==100093== 
==100093== 250 (112 direct, 138 indirect) bytes in 1 blocks are definitely lost in loss record 6,987 of 7,502
==100093==    at 0x4846FA3: operator new(unsigned long) (in /usr/libexec/valgrind/vgpreload_memcheck-amd64-linux.so)
==100093==    by 0x2069EC: C_A_T_C_H_T_E_S_T_0() (test_player.cpp:64)
==100093==    by 0x14415D: Catch::TestInvokerAsFunction::invoke() const (catch.hpp:14330)
==100093==    by 0x143394: Catch::TestCase::invoke() const (catch.hpp:14169)
==100093==    by 0x13D69F: Catch::RunContext::invokeActiveTestCase() (catch.hpp:13025)
==100093==    by 0x13D3C4: Catch::RunContext::runCurrentTest(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >&) (catch.hpp:12998)
==100093==    by 0x13BE50: Catch::RunContext::runTest(Catch::TestCase const&) (catch.hpp:12759)
==100093==    by 0x13F045: Catch::(anonymous namespace)::TestGroup::execute() (catch.hpp:13352)
==100093==    by 0x1404AC: Catch::Session::runInternal() (catch.hpp:13562)
==100093==    by 0x140186: Catch::Session::run() (catch.hpp:13518)
==100093==    by 0x1812EE: int Catch::Session::run<char>(int, char const* const*) (catch.hpp:13236)
==100093==    by 0x15735F: main (main.cpp:8)
==100093== 
==100093== 288 bytes in 12 blocks are definitely lost in loss record 7,049 of 7,502
==100093==    at 0x4846FA3: operator new(unsigned long) (in /usr/libexec/valgrind/vgpreload_memcheck-amd64-linux.so)
==100093==    by 0x22FF1F: Deck::Deck() (deck.cpp:33)
==100093==    by 0x1FBD15: C_A_T_C_H_T_E_S_T_0() (test_deck.cpp:135)
==100093==    by 0x14415D: Catch::TestInvokerAsFunction::invoke() const (catch.hpp:14330)
==100093==    by 0x143394: Catch::TestCase::invoke() const (catch.hpp:14169)
==100093==    by 0x13D69F: Catch::RunContext::invokeActiveTestCase() (catch.hpp:13025)
==100093==    by 0x13D3C4: Catch::RunContext::runCurrentTest(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >&) (catch.hpp:12998)
==100093==    by 0x13BE50: Catch::RunContext::runTest(Catch::TestCase const&) (catch.hpp:12759)
==100093==    by 0x13F045: Catch::(anonymous namespace)::TestGroup::execute() (catch.hpp:13352)
==100093==    by 0x1404AC: Catch::Session::runInternal() (catch.hpp:13562)
==100093==    by 0x140186: Catch::Session::run() (catch.hpp:13518)
==100093==    by 0x1812EE: int Catch::Session::run<char>(int, char const* const*) (catch.hpp:13236)
==100093== 
==100093== 288 bytes in 12 blocks are definitely lost in loss record 7,050 of 7,502
==100093==    at 0x4846FA3: operator new(unsigned long) (in /usr/libexec/valgrind/vgpreload_memcheck-amd64-linux.so)
==100093==    by 0x22FF5B: Deck::Deck() (deck.cpp:34)
==100093==    by 0x1FBD15: C_A_T_C_H_T_E_S_T_0() (test_deck.cpp:135)
==100093==    by 0x14415D: Catch::TestInvokerAsFunction::invoke() const (catch.hpp:14330)
==100093==    by 0x143394: Catch::TestCase::invoke() const (catch.hpp:14169)
==100093==    by 0x13D69F: Catch::RunContext::invokeActiveTestCase() (catch.hpp:13025)
==100093==    by 0x13D3C4: Catch::RunContext::runCurrentTest(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >&) (catch.hpp:12998)
==100093==    by 0x13BE50: Catch::RunContext::runTest(Catch::TestCase const&) (catch.hpp:12759)
==100093==    by 0x13F045: Catch::(anonymous namespace)::TestGroup::execute() (catch.hpp:13352)
==100093==    by 0x1404AC: Catch::Session::runInternal() (catch.hpp:13562)
==100093==    by 0x140186: Catch::Session::run() (catch.hpp:13518)
==100093==    by 0x1812EE: int Catch::Session::run<char>(int, char const* const*) (catch.hpp:13236)
==100093== 
==100093== 288 bytes in 12 blocks are definitely lost in loss record 7,051 of 7,502
==100093==    at 0x4846FA3: operator new(unsigned long) (in /usr/libexec/valgrind/vgpreload_memcheck-amd64-linux.so)
==100093==    by 0x22FF1F: Deck::Deck() (deck.cpp:33)
==100093==    by 0x1FC76C: C_A_T_C_H_T_E_S_T_0() (test_deck.cpp:206)
==100093==    by 0x14415D: Catch::TestInvokerAsFunction::invoke() const (catch.hpp:14330)
==100093==    by 0x143394: Catch::TestCase::invoke() const (catch.hpp:14169)
==100093==    by 0x13D69F: Catch::RunContext::invokeActiveTestCase() (catch.hpp:13025)
==100093==    by 0x13D3C4: Catch::RunContext::runCurrentTest(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >&) (catch.hpp:12998)
==100093==    by 0x13BE50: Catch::RunContext::runTest(Catch::TestCase const&) (catch.hpp:12759)
==100093==    by 0x13F045: Catch::(anonymous namespace)::TestGroup::execute() (catch.hpp:13352)
==100093==    by 0x1404AC: Catch::Session::runInternal() (catch.hpp:13562)
==100093==    by 0x140186: Catch::Session::run() (catch.hpp:13518)
==100093==    by 0x1812EE: int Catch::Session::run<char>(int, char const* const*) (catch.hpp:13236)
==100093== 
==100093== 288 bytes in 12 blocks are definitely lost in loss record 7,052 of 7,502
==100093==    at 0x4846FA3: operator new(unsigned long) (in /usr/libexec/valgrind/vgpreload_memcheck-amd64-linux.so)
==100093==    by 0x22FF5B: Deck::Deck() (deck.cpp:34)
==100093==    by 0x1FC76C: C_A_T_C_H_T_E_S_T_0() (test_deck.cpp:206)
==100093==    by 0x14415D: Catch::TestInvokerAsFunction::invoke() const (catch.hpp:14330)
==100093==    by 0x143394: Catch::TestCase::invoke() const (catch.hpp:14169)
==100093==    by 0x13D69F: Catch::RunContext::invokeActiveTestCase() (catch.hpp:13025)
==100093==    by 0x13D3C4: Catch::RunContext::runCurrentTest(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >&) (catch.hpp:12998)
==100093==    by 0x13BE50: Catch::RunContext::runTest(Catch::TestCase const&) (catch.hpp:12759)
==100093==    by 0x13F045: Catch::(anonymous namespace)::TestGroup::execute() (catch.hpp:13352)
==100093==    by 0x1404AC: Catch::Session::runInternal() (catch.hpp:13562)
==100093==    by 0x140186: Catch::Session::run() (catch.hpp:13518)
==100093==    by 0x1812EE: int Catch::Session::run<char>(int, char const* const*) (catch.hpp:13236)
==100093== 
==100093== 288 bytes in 12 blocks are definitely lost in loss record 7,053 of 7,502
==100093==    at 0x4846FA3: operator new(unsigned long) (in /usr/libexec/valgrind/vgpreload_memcheck-amd64-linux.so)
==100093==    by 0x22FF1F: Deck::Deck() (deck.cpp:33)
==100093==    by 0x1FC9B7: C_A_T_C_H_T_E_S_T_0() (test_deck.cpp:222)
==100093==    by 0x14415D: Catch::TestInvokerAsFunction::invoke() const (catch.hpp:14330)
==100093==    by 0x143394: Catch::TestCase::invoke() const (catch.hpp:14169)
==100093==    by 0x13D69F: Catch::RunContext::invokeActiveTestCase() (catch.hpp:13025)
==100093==    by 0x13D3C4: Catch::RunContext::runCurrentTest(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >&) (catch.hpp:12998)
==100093==    by 0x13BE50: Catch::RunContext::runTest(Catch::TestCase const&) (catch.hpp:12759)
==100093==    by 0x13F045: Catch::(anonymous namespace)::TestGroup::execute() (catch.hpp:13352)
==100093==    by 0x1404AC: Catch::Session::runInternal() (catch.hpp:13562)
==100093==    by 0x140186: Catch::Session::run() (catch.hpp:13518)
==100093==    by 0x1812EE: int Catch::Session::run<char>(int, char const* const*) (catch.hpp:13236)
==100093== 
==100093== 288 bytes in 12 blocks are definitely lost in loss record 7,054 of 7,502
==100093==    at 0x4846FA3: operator new(unsigned long) (in /usr/libexec/valgrind/vgpreload_memcheck-amd64-linux.so)
==100093==    by 0x22FF5B: Deck::Deck() (deck.cpp:34)
==100093==    by 0x1FC9B7: C_A_T_C_H_T_E_S_T_0() (test_deck.cpp:222)
==100093==    by 0x14415D: Catch::TestInvokerAsFunction::invoke() const (catch.hpp:14330)
==100093==    by 0x143394: Catch::TestCase::invoke() const (catch.hpp:14169)
==100093==    by 0x13D69F: Catch::RunContext::invokeActiveTestCase() (catch.hpp:13025)
==100093==    by 0x13D3C4: Catch::RunContext::runCurrentTest(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >&) (catch.hpp:12998)
==100093==    by 0x13BE50: Catch::RunContext::runTest(Catch::TestCase const&) (catch.hpp:12759)
==100093==    by 0x13F045: Catch::(anonymous namespace)::TestGroup::execute() (catch.hpp:13352)
==100093==    by 0x1404AC: Catch::Session::runInternal() (catch.hpp:13562)
==100093==    by 0x140186: Catch::Session::run() (catch.hpp:13518)
==100093==    by 0x1812EE: int Catch::Session::run<char>(int, char const* const*) (catch.hpp:13236)
==100093== 
==100093== 288 bytes in 12 blocks are definitely lost in loss record 7,055 of 7,502
==100093==    at 0x4846FA3: operator new(unsigned long) (in /usr/libexec/valgrind/vgpreload_memcheck-amd64-linux.so)
==100093==    by 0x22FF1F: Deck::Deck() (deck.cpp:33)
==100093==    by 0x20CD6F: C_A_T_C_H_T_E_S_T_0() (test_game.cpp:24)
==100093==    by 0x14415D: Catch::TestInvokerAsFunction::invoke() const (catch.hpp:14330)
==100093==    by 0x143394: Catch::TestCase::invoke() const (catch.hpp:14169)
==100093==    by 0x13D69F: Catch::RunContext::invokeActiveTestCase() (catch.hpp:13025)
==100093==    by 0x13D3C4: Catch::RunContext::runCurrentTest(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >&) (catch.hpp:12998)
==100093==    by 0x13BE50: Catch::RunContext::runTest(Catch::TestCase const&) (catch.hpp:12759)
==100093==    by 0x13F045: Catch::(anonymous namespace)::TestGroup::execute() (catch.hpp:13352)
==100093==    by 0x1404AC: Catch::Session::runInternal() (catch.hpp:13562)
==100093==    by 0x140186: Catch::Session::run() (catch.hpp:13518)
==100093==    by 0x1812EE: int Catch::Session::run<char>(int, char const* const*) (catch.hpp:13236)
==100093== 
==100093== 288 bytes in 12 blocks are definitely lost in loss record 7,056 of 7,502
==100093==    at 0x4846FA3: operator new(unsigned long) (in /usr/libexec/valgrind/vgpreload_memcheck-amd64-linux.so)
==100093==    by 0x22FF5B: Deck::Deck() (deck.cpp:34)
==100093==    by 0x20CD6F: C_A_T_C_H_T_E_S_T_0() (test_game.cpp:24)
==100093==    by 0x14415D: Catch::TestInvokerAsFunction::invoke() const (catch.hpp:14330)
==100093==    by 0x143394: Catch::TestCase::invoke() const (catch.hpp:14169)
==100093==    by 0x13D69F: Catch::RunContext::invokeActiveTestCase() (catch.hpp:13025)
==100093==    by 0x13D3C4: Catch::RunContext::runCurrentTest(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >&) (catch.hpp:12998)
==100093==    by 0x13BE50: Catch::RunContext::runTest(Catch::TestCase const&) (catch.hpp:12759)
==100093==    by 0x13F045: Catch::(anonymous namespace)::TestGroup::execute() (catch.hpp:13352)
==100093==    by 0x1404AC: Catch::Session::runInternal() (catch.hpp:13562)
==100093==    by 0x140186: Catch::Session::run() (catch.hpp:13518)
==100093==    by 0x1812EE: int Catch::Session::run<char>(int, char const* const*) (catch.hpp:13236)
==100093== 
==100093== 288 bytes in 12 blocks are definitely lost in loss record 7,057 of 7,502
==100093==    at 0x4846FA3: operator new(unsigned long) (in /usr/libexec/valgrind/vgpreload_memcheck-amd64-linux.so)
==100093==    by 0x22FF1F: Deck::Deck() (deck.cpp:33)
==100093==    by 0x20D4E3: C_A_T_C_H_T_E_S_T_0() (test_game.cpp:78)
==100093==    by 0x14415D: Catch::TestInvokerAsFunction::invoke() const (catch.hpp:14330)
==100093==    by 0x143394: Catch::TestCase::invoke() const (catch.hpp:14169)
==100093==    by 0x13D69F: Catch::RunContext::invokeActiveTestCase() (catch.hpp:13025)
==100093==    by 0x13D3C4: Catch::RunContext::runCurrentTest(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >&) (catch.hpp:12998)
==100093==    by 0x13BE50: Catch::RunContext::runTest(Catch::TestCase const&) (catch.hpp:12759)
==100093==    by 0x13F045: Catch::(anonymous namespace)::TestGroup::execute() (catch.hpp:13352)
==100093==    by 0x1404AC: Catch::Session::runInternal() (catch.hpp:13562)
==100093==    by 0x140186: Catch::Session::run() (catch.hpp:13518)
==100093==    by 0x1812EE: int Catch::Session::run<char>(int, char const* const*) (catch.hpp:13236)
==100093== 
==100093== 288 bytes in 12 blocks are definitely lost in loss record 7,058 of 7,502
==100093==    at 0x4846FA3: operator new(unsigned long) (in /usr/libexec/valgrind/vgpreload_memcheck-amd64-linux.so)
==100093==    by 0x22FF5B: Deck::Deck() (deck.cpp:34)
==100093==    by 0x20D4E3: C_A_T_C_H_T_E_S_T_0() (test_game.cpp:78)
==100093==    by 0x14415D: Catch::TestInvokerAsFunction::invoke() const (catch.hpp:14330)
==100093==    by 0x143394: Catch::TestCase::invoke() const (catch.hpp:14169)
==100093==    by 0x13D69F: Catch::RunContext::invokeActiveTestCase() (catch.hpp:13025)
==100093==    by 0x13D3C4: Catch::RunContext::runCurrentTest(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >&) (catch.hpp:12998)
==100093==    by 0x13BE50: Catch::RunContext::runTest(Catch::TestCase const&) (catch.hpp:12759)
==100093==    by 0x13F045: Catch::(anonymous namespace)::TestGroup::execute() (catch.hpp:13352)
==100093==    by 0x1404AC: Catch::Session::runInternal() (catch.hpp:13562)
==100093==    by 0x140186: Catch::Session::run() (catch.hpp:13518)
==100093==    by 0x1812EE: int Catch::Session::run<char>(int, char const* const*) (catch.hpp:13236)
==100093== 
==100093== 864 bytes in 36 blocks are definitely lost in loss record 7,284 of 7,502
==100093==    at 0x4846FA3: operator new(unsigned long) (in /usr/libexec/valgrind/vgpreload_memcheck-amd64-linux.so)
==100093==    by 0x22FE5B: Deck::Deck() (deck.cpp:26)
==100093==    by 0x1FBD15: C_A_T_C_H_T_E_S_T_0() (test_deck.cpp:135)
==100093==    by 0x14415D: Catch::TestInvokerAsFunction::invoke() const (catch.hpp:14330)
==100093==    by 0x143394: Catch::TestCase::invoke() const (catch.hpp:14169)
==100093==    by 0x13D69F: Catch::RunContext::invokeActiveTestCase() (catch.hpp:13025)
==100093==    by 0x13D3C4: Catch::RunContext::runCurrentTest(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >&) (catch.hpp:12998)
==100093==    by 0x13BE50: Catch::RunContext::runTest(Catch::TestCase const&) (catch.hpp:12759)
==100093==    by 0x13F045: Catch::(anonymous namespace)::TestGroup::execute() (catch.hpp:13352)
==100093==    by 0x1404AC: Catch::Session::runInternal() (catch.hpp:13562)
==100093==    by 0x140186: Catch::Session::run() (catch.hpp:13518)
==100093==    by 0x1812EE: int Catch::Session::run<char>(int, char const* const*) (catch.hpp:13236)
==100093== 
==100093== 864 bytes in 36 blocks are definitely lost in loss record 7,285 of 7,502
==100093==    at 0x4846FA3: operator new(unsigned long) (in /usr/libexec/valgrind/vgpreload_memcheck-amd64-linux.so)
==100093==    by 0x22FE5B: Deck::Deck() (deck.cpp:26)
==100093==    by 0x1FC76C: C_A_T_C_H_T_E_S_T_0() (test_deck.cpp:206)
==100093==    by 0x14415D: Catch::TestInvokerAsFunction::invoke() const (catch.hpp:14330)
==100093==    by 0x143394: Catch::TestCase::invoke() const (catch.hpp:14169)
==100093==    by 0x13D69F: Catch::RunContext::invokeActiveTestCase() (catch.hpp:13025)
==100093==    by 0x13D3C4: Catch::RunContext::runCurrentTest(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >&) (catch.hpp:12998)
==100093==    by 0x13BE50: Catch::RunContext::runTest(Catch::TestCase const&) (catch.hpp:12759)
==100093==    by 0x13F045: Catch::(anonymous namespace)::TestGroup::execute() (catch.hpp:13352)
==100093==    by 0x1404AC: Catch::Session::runInternal() (catch.hpp:13562)
==100093==    by 0x140186: Catch::Session::run() (catch.hpp:13518)
==100093==    by 0x1812EE: int Catch::Session::run<char>(int, char const* const*) (catch.hpp:13236)
==100093== 
==100093== 864 bytes in 36 blocks are definitely lost in loss record 7,286 of 7,502
==100093==    at 0x4846FA3: operator new(unsigned long) (in /usr/libexec/valgrind/vgpreload_memcheck-amd64-linux.so)
==100093==    by 0x22FE5B: Deck::Deck() (deck.cpp:26)
==100093==    by 0x1FC9B7: C_A_T_C_H_T_E_S_T_0() (test_deck.cpp:222)
==100093==    by 0x14415D: Catch::TestInvokerAsFunction::invoke() const (catch.hpp:14330)
==100093==    by 0x143394: Catch::TestCase::invoke() const (catch.hpp:14169)
==100093==    by 0x13D69F: Catch::RunContext::invokeActiveTestCase() (catch.hpp:13025)
==100093==    by 0x13D3C4: Catch::RunContext::runCurrentTest(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >&) (catch.hpp:12998)
==100093==    by 0x13BE50: Catch::RunContext::runTest(Catch::TestCase const&) (catch.hpp:12759)
==100093==    by 0x13F045: Catch::(anonymous namespace)::TestGroup::execute() (catch.hpp:13352)
==100093==    by 0x1404AC: Catch::Session::runInternal() (catch.hpp:13562)
==100093==    by 0x140186: Catch::Session::run() (catch.hpp:13518)
==100093==    by 0x1812EE: int Catch::Session::run<char>(int, char const* const*) (catch.hpp:13236)
==100093== 
==100093== 864 bytes in 36 blocks are definitely lost in loss record 7,287 of 7,502
==100093==    at 0x4846FA3: operator new(unsigned long) (in /usr/libexec/valgrind/vgpreload_memcheck-amd64-linux.so)
==100093==    by 0x22FE5B: Deck::Deck() (deck.cpp:26)
==100093==    by 0x20CD6F: C_A_T_C_H_T_E_S_T_0() (test_game.cpp:24)
==100093==    by 0x14415D: Catch::TestInvokerAsFunction::invoke() const (catch.hpp:14330)
==100093==    by 0x143394: Catch::TestCase::invoke() const (catch.hpp:14169)
==100093==    by 0x13D69F: Catch::RunContext::invokeActiveTestCase() (catch.hpp:13025)
==100093==    by 0x13D3C4: Catch::RunContext::runCurrentTest(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >&) (catch.hpp:12998)
==100093==    by 0x13BE50: Catch::RunContext::runTest(Catch::TestCase const&) (catch.hpp:12759)
==100093==    by 0x13F045: Catch::(anonymous namespace)::TestGroup::execute() (catch.hpp:13352)
==100093==    by 0x1404AC: Catch::Session::runInternal() (catch.hpp:13562)
==100093==    by 0x140186: Catch::Session::run() (catch.hpp:13518)
==100093==    by 0x1812EE: int Catch::Session::run<char>(int, char const* const*) (catch.hpp:13236)
==100093== 
==100093== 864 bytes in 36 blocks are definitely lost in loss record 7,288 of 7,502
==100093==    at 0x4846FA3: operator new(unsigned long) (in /usr/libexec/valgrind/vgpreload_memcheck-amd64-linux.so)
==100093==    by 0x22FE5B: Deck::Deck() (deck.cpp:26)
==100093==    by 0x20D4E3: C_A_T_C_H_T_E_S_T_0() (test_game.cpp:78)
==100093==    by 0x14415D: Catch::TestInvokerAsFunction::invoke() const (catch.hpp:14330)
==100093==    by 0x143394: Catch::TestCase::invoke() const (catch.hpp:14169)
==100093==    by 0x13D69F: Catch::RunContext::invokeActiveTestCase() (catch.hpp:13025)
==100093==    by 0x13D3C4: Catch::RunContext::runCurrentTest(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >&) (catch.hpp:12998)
==100093==    by 0x13BE50: Catch::RunContext::runTest(Catch::TestCase const&) (catch.hpp:12759)
==100093==    by 0x13F045: Catch::(anonymous namespace)::TestGroup::execute() (catch.hpp:13352)
==100093==    by 0x1404AC: Catch::Session::runInternal() (catch.hpp:13562)
==100093==    by 0x140186: Catch::Session::run() (catch.hpp:13518)
==100093==    by 0x1812EE: int Catch::Session::run<char>(int, char const* const*) (catch.hpp:13236)
==100093== 
==100093== 936 bytes in 39 blocks are definitely lost in loss record 7,295 of 7,502
==100093==    at 0x4846FA3: operator new(unsigned long) (in /usr/libexec/valgrind/vgpreload_memcheck-amd64-linux.so)
==100093==    by 0x22FE99: Deck::Deck() (deck.cpp:27)
==100093==    by 0x20D4E3: C_A_T_C_H_T_E_S_T_0() (test_game.cpp:78)
==100093==    by 0x14415D: Catch::TestInvokerAsFunction::invoke() const (catch.hpp:14330)
==100093==    by 0x143394: Catch::TestCase::invoke() const (catch.hpp:14169)
==100093==    by 0x13D69F: Catch::RunContext::invokeActiveTestCase() (catch.hpp:13025)
==100093==    by 0x13D3C4: Catch::RunContext::runCurrentTest(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >&) (catch.hpp:12998)
==100093==    by 0x13BE50: Catch::RunContext::runTest(Catch::TestCase const&) (catch.hpp:12759)
==100093==    by 0x13F045: Catch::(anonymous namespace)::TestGroup::execute() (catch.hpp:13352)
==100093==    by 0x1404AC: Catch::Session::runInternal() (catch.hpp:13562)
==100093==    by 0x140186: Catch::Session::run() (catch.hpp:13518)
==100093==    by 0x1812EE: int Catch::Session::run<char>(int, char const* const*) (catch.hpp:13236)
==100093== 
==100093== 960 bytes in 40 blocks are definitely lost in loss record 7,300 of 7,502
==100093==    at 0x4846FA3: operator new(unsigned long) (in /usr/libexec/valgrind/vgpreload_memcheck-amd64-linux.so)
==100093==    by 0x22FE99: Deck::Deck() (deck.cpp:27)
==100093==    by 0x1FBD15: C_A_T_C_H_T_E_S_T_0() (test_deck.cpp:135)
==100093==    by 0x14415D: Catch::TestInvokerAsFunction::invoke() const (catch.hpp:14330)
==100093==    by 0x143394: Catch::TestCase::invoke() const (catch.hpp:14169)
==100093==    by 0x13D69F: Catch::RunContext::invokeActiveTestCase() (catch.hpp:13025)
==100093==    by 0x13D3C4: Catch::RunContext::runCurrentTest(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >&) (catch.hpp:12998)
==100093==    by 0x13BE50: Catch::RunContext::runTest(Catch::TestCase const&) (catch.hpp:12759)
==100093==    by 0x13F045: Catch::(anonymous namespace)::TestGroup::execute() (catch.hpp:13352)
==100093==    by 0x1404AC: Catch::Session::runInternal() (catch.hpp:13562)
==100093==    by 0x140186: Catch::Session::run() (catch.hpp:13518)
==100093==    by 0x1812EE: int Catch::Session::run<char>(int, char const* const*) (catch.hpp:13236)
==100093== 
==100093== 960 bytes in 40 blocks are definitely lost in loss record 7,301 of 7,502
==100093==    at 0x4846FA3: operator new(unsigned long) (in /usr/libexec/valgrind/vgpreload_memcheck-amd64-linux.so)
==100093==    by 0x22FE99: Deck::Deck() (deck.cpp:27)
==100093==    by 0x1FC76C: C_A_T_C_H_T_E_S_T_0() (test_deck.cpp:206)
==100093==    by 0x14415D: Catch::TestInvokerAsFunction::invoke() const (catch.hpp:14330)
==100093==    by 0x143394: Catch::TestCase::invoke() const (catch.hpp:14169)
==100093==    by 0x13D69F: Catch::RunContext::invokeActiveTestCase() (catch.hpp:13025)
==100093==    by 0x13D3C4: Catch::RunContext::runCurrentTest(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >&) (catch.hpp:12998)
==100093==    by 0x13BE50: Catch::RunContext::runTest(Catch::TestCase const&) (catch.hpp:12759)
==100093==    by 0x13F045: Catch::(anonymous namespace)::TestGroup::execute() (catch.hpp:13352)
==100093==    by 0x1404AC: Catch::Session::runInternal() (catch.hpp:13562)
==100093==    by 0x140186: Catch::Session::run() (catch.hpp:13518)
==100093==    by 0x1812EE: int Catch::Session::run<char>(int, char const* const*) (catch.hpp:13236)
==100093== 
==100093== 960 bytes in 40 blocks are definitely lost in loss record 7,302 of 7,502
==100093==    at 0x4846FA3: operator new(unsigned long) (in /usr/libexec/valgrind/vgpreload_memcheck-amd64-linux.so)
==100093==    by 0x22FE99: Deck::Deck() (deck.cpp:27)
==100093==    by 0x1FC9B7: C_A_T_C_H_T_E_S_T_0() (test_deck.cpp:222)
==100093==    by 0x14415D: Catch::TestInvokerAsFunction::invoke() const (catch.hpp:14330)
==100093==    by 0x143394: Catch::TestCase::invoke() const (catch.hpp:14169)
==100093==    by 0x13D69F: Catch::RunContext::invokeActiveTestCase() (catch.hpp:13025)
==100093==    by 0x13D3C4: Catch::RunContext::runCurrentTest(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >&) (catch.hpp:12998)
==100093==    by 0x13BE50: Catch::RunContext::runTest(Catch::TestCase const&) (catch.hpp:12759)
==100093==    by 0x13F045: Catch::(anonymous namespace)::TestGroup::execute() (catch.hpp:13352)
==100093==    by 0x1404AC: Catch::Session::runInternal() (catch.hpp:13562)
==100093==    by 0x140186: Catch::Session::run() (catch.hpp:13518)
==100093==    by 0x1812EE: int Catch::Session::run<char>(int, char const* const*) (catch.hpp:13236)
==100093== 
==100093== 960 bytes in 40 blocks are definitely lost in loss record 7,303 of 7,502
==100093==    at 0x4846FA3: operator new(unsigned long) (in /usr/libexec/valgrind/vgpreload_memcheck-amd64-linux.so)
==100093==    by 0x22FE99: Deck::Deck() (deck.cpp:27)
==100093==    by 0x20CD6F: C_A_T_C_H_T_E_S_T_0() (test_game.cpp:24)
==100093==    by 0x14415D: Catch::TestInvokerAsFunction::invoke() const (catch.hpp:14330)
==100093==    by 0x143394: Catch::TestCase::invoke() const (catch.hpp:14169)
==100093==    by 0x13D69F: Catch::RunContext::invokeActiveTestCase() (catch.hpp:13025)
==100093==    by 0x13D3C4: Catch::RunContext::runCurrentTest(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >&) (catch.hpp:12998)
==100093==    by 0x13BE50: Catch::RunContext::runTest(Catch::TestCase const&) (catch.hpp:12759)
==100093==    by 0x13F045: Catch::(anonymous namespace)::TestGroup::execute() (catch.hpp:13352)
==100093==    by 0x1404AC: Catch::Session::runInternal() (catch.hpp:13562)
==100093==    by 0x140186: Catch::Session::run() (catch.hpp:13518)
==100093==    by 0x1812EE: int Catch::Session::run<char>(int, char const* const*) (catch.hpp:13236)
==100093== 
==100093== LEAK SUMMARY:
==100093==    definitely lost: 13,312 bytes in 549 blocks
==100093==    indirectly lost: 386 bytes in 12 blocks
==100093==      possibly lost: 0 bytes in 0 blocks
==100093==    still reachable: 690 bytes in 8 blocks
==100093==                       of which reachable via heuristic:
==100093==                         newarray           : 760 bytes in 5 blocks
==100093==         suppressed: 2,334,212 bytes in 22,869 blocks
==100093== 
==100093== For lists of detected and suppressed errors, rerun with: -s
==100093== ERROR SUMMARY: 42 errors from 42 contexts (suppressed: 55 from 55)
